app/va/va_profile/va_profile_client.py

```diff
@@ -136,6 +156,94 @@ def get_email(self, va_profile_id: RecipientIdentifier) -> str:
         self.statsd_client.incr('clients.va-profile.get-email.failure')
         self._raise_no_contact_info_exception(self.EMAIL_BIO_TYPE, va_profile_id, contact_info.get(self.TX_AUDIT_ID))
 
+    def get_telephone_with_permission(
+        self,
+        va_profile_id: RecipientIdentifier,
+        default_send=True,
+    ) -> VAProfileResult:
+        """
+        Retrieve the telephone number from the profile information for a given VA profile ID.
+
+        Args:
+            va_profile_id (RecipientIdentifier): The VA profile ID to retrieve the telephone number for.
+
+        Returns:
+            VAProfileResults: The result data.
+            Property recipient is the telephone number retrieved from the VA Profile service.
+            Property communication_allowed is true when VA Profile service indicates that the recipient has allowed communication.
+            Property permission_message may contain an error message if the permission check encountered an exception.
+        """
+        profile = self.get_profile(va_profile_id)
+        communication_allowed = True
+        permission_message = None
+        try:
+            communication_allowed = self.get_is_communication_allowed_from_profile(profile, CommunicationChannel.TEXT)
+        except CommunicationItemNotFoundException:
+            self.logger.info('Communication item for recipient %s not found', va_profile_id)
+            communication_allowed = default_send
+            permission_message = f'V3 Profile - No recipient opt-in found for explicit preference, falling back to default send: {default_send} (Recipient Identifier {va_profile_id})'
+
+        contact_info: ContactInformation = profile.get('contactInformation', {})
+
+        telephones: List[Telephone] = contact_info.get(self.PHONE_BIO_TYPE, [])
+        sorted_telephones = sorted(
+            [phone for phone in telephones if phone['phoneType'] == PhoneNumberType.MOBILE.value],
```

Comment:
I thought we weren't going to filter on the PhoneNumberType because someone could have just put their number in the "HOME" category or something. This could result in veterans not receiving notifications simply because they didn't put their number in the correct field.

Reply:
Hmm, I'm not sure about that.  We do have the [follow-up work](https://github.com/department-of-veterans-affairs/va-iir/issues/900) to verify that the number is indeed a mobile number that we can send sms messages to it. As it currently stands, a person could have included a landline telephone in a "mobile telephone field" - resulting in a delivery failure.

I don't think we've investigated the possibility of a person entering a mobile number in a landline phone input field.  If we allowed all phone numbers, then we would see a dramatic increase in the sms send failure rate, because we would attempt to send sms messages to whatever the last phone number they entered was (which could be a landline phone).

The linked ticket would need to be deployed before we could consider a strategy to implement this.  In that case, I would still suggest we leave the code as-is, so we don't kick off a stream of destined-to-fail SMS requests

Reply:
okay, we can revisit this in the follow-up ticket.

Reply:
we can also leave the feature flag OFF in Prod until more work is completed and then there is no risk of dropping real messages for these reasons.